emails/models.py

```diff
@@ -784,6 +790,8 @@ def save(self, *args, **kwargs) -> None:
             self.block_list_emails = False
         if not user_profile.server_storage:
             self.description = ""
+        user_profile.last_engagement = datetime.now(timezone.utc)
+        user_profile.save()
```

Comment:
Do you intend that every change to a domain mask causes the user to be marked as engaged?

Reply:
Oh good catch let me clarify that.

Reply:
Moved this up into the `if self._state.adding` block, because we only want to update the field when the user adds a domain address.